scripts/build-tools.sh

```diff
@@ -5,69 +5,127 @@
 # fail immediately on any errors
 set -e
 
-
 print_usage()
 {
         cat <<EOFUSAGE
-usage: $0 [-t|-f]
+usage: $0 [-c|-f|-h|-l|-p|-t|-T]
+       [-c] Build ctl
+       [-f] Build fuzzer
+       [-h] Display help
+       [-l] Build logger
+       [-p] Build probes
        [-t] Build test topologies
-       [-f] Build fuzzer"
+       [-T] Build topologies
+       [-n] No build, only cmake
```

Comment:
Couldn't decide on a better single letter option to represent "No build, only cmake". Open to suggestions.

Reply:
Could you change `Build ctl` to `Re-build ctl` etc.?

The `[ ]` brackets are not needed: they mean something is optional which makes sense only in the SYNOPSIS. Look at `man tar` and `tar --help` for a good example.

> Couldn't decide on a better single letter option to represent "No build, only cmake". Open to suggestions.

`-n` is fine, maybe you'll prefer this:
```
usage: $0 [OPTION...]
...
  -C  No build, only CMake re-configuration. Ignored when building specific tool(s).
```
The "ignored" part I added is IMHO important.

Reply:
Although it is not ignored when building specific tool. If I do `./scripts/build-tools.sh -c -n` it will only generate the configuration and print build instructions.